homeassistant/components/supla/__init__.py

```diff
@@ -98,7 +98,7 @@ async def discover_devices(hass, hass_config):
 
     Currently it is only run at startup.
     """
-    component_configs = {}
+    component_configs: dict[Platform, list[dict]] = {}
```

Comment:
you should use the enums also in line 30 in `SUPLA_FUNCTION_HA_CMP_MAP`
```diff
SUPLA_FUNCTION_HA_CMP_MAP = {
-    "CONTROLLINGTHEROLLERSHUTTER": "cover",
+    "CONTROLLINGTHEROLLERSHUTTER": Platform.COVER,
-    "CONTROLLINGTHEGATE": "cover",
+    "CONTROLLINGTHEGATE": Platform.COVER,
-    "LIGHTSWITCH": "switch",
+    "LIGHTSWITCH": Platform.SWITCH,
}
```

Reply:
@mib1185 this should be ok now.